### ISSUE TITLE
ReactWrapper::setState() expects a function as its second argument

### ISSUE THREAD

**Issue by ronanmulvaney** [airbnb/enzyme#1764]
We're still hitting a problem with an upgrade to 3.4.3 with calls in some context to setState for tests that were working on 3.4.1.

The remaining failure is:
```
    TypeError: ReactWrapper::setState() expects a function as its second argument

      389 |    */
      390 |   handleBackCallback() {
    > 391 |     this.setState({ error: false }, this.getPageDetails({ page: 'previous' }));
          |          ^
      392 |   }
      393 |
      394 |   /**

      at ShallowWrapper.setState (packages/test-helpers/node_modules/enzyme/build/ShallowWrapper.js:508:17)
      at ApplicationForm.ShallowWrapper.instance.setState (packages/test-helpers/node_modules/enzyme/build/ShallowWrapper.js:198:33)
      at ApplicationForm.setState (packages/universal-access-ui/src/features/ApplicationForm/application-form/ApplicationForm.js:391:10)
      at Object.handleBackCallback (packages/universal-access-ui/src/features/ApplicationForm/application-form/ApplicationForm_functional.test.js:131:14)
```

The second argument is of course a function but whatever has been changing with setState in enzyme is affecting this.

**Comment by Tinusw** [comment]
I've got a similar issue where some setState calls throw errors during shallow rendered tests.

This was working in `3.4.1`

```
Method “setState” is only meant to be run on a single node. undefined found instead.
 
componentDidMount(){
   this.setState({ valid: this.props.oldEmail != this.state.email })
}
```

I believe this is related to the multiple changes made in `3.4.3` to `Shallow` -> `setState`

**Comment by ronanmulvaney** [comment]
@Tinusw As I understand the sequence of changes, 3.4.2 broke a few things in relation to setState that were believed corrected in 3.4.3 with #1756 which I think matches your failure.

This is just to highlight some remaining use cases being affected.

As you mention, sticking on 3.4.1 avoids.

**Comment by ronanmulvaney** [comment]
Yes, apologies with the other failures yesterday hadn't looked closely at it since it was passing an expect getPageDetails to be fired which of course it would be but not validly!!
